### ISSUE TITLE
tensorflow crashed when use image.transform function

### ISSUE THREAD

**Issue by zhuimeng999** [tensorflow/addons#2152]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): arch Linux latest and Ubuntu
- TensorFlow version and how it was installed (source or binary): binary on arch, source on ubuntu 
- TensorFlow-Addons version and how it was installed (source or binary): source version 0.11.2
- Python version: 3.8 on arch, 3.6 on ubuntu
- Is GPU used? (yes/no): yes

**Describe the bug**

tensorflow crashed when use image.transform module

**Code to reproduce the issue**

import tensorflow as tf
import tensorflow_addons as tfa
import numpy as np

if __name__ == '__main__':
    src = tf.Variable(np.random.random((1, 100, 100, 3)))
    dst = np.random.randint(0, 3, (1, 100, 100))

    with tf.GradientTape() as tape:
        out = tfa.image.transform(src, np.random.rand(8))
        loss = tf.nn.sparse_softmax_cross_entropy_with_logits(dst, out)
        loss = tf.reduce_mean(loss)
    tape.gradient(loss, src)

**Other info / logs**
the output log may display failed at

2020-09-07 14:07:25.563878: F tensorflow/core/kernels/cuda_solvers.cc:120] Check failed: cublasCreate(&cublas_handle) == CUBLAS_STATUS_SUCCESS Failed to create cuBlas instance.
[log.txt](https://github.com/tensorflow/addons/files/5181375/log.txt)

**Comment by WindQAQ** [comment]
This is probably your cuda error. Can you paste the detailed log?

https://colab.research.google.com/drive/1x70ApzFL01APODQxlcnvVPIJJVUQR3DI?usp=sharing

**Comment by zhuimeng999** [comment]
may be, i think it is a tensorflow issue before, bug when is replace the code about image.transform, the problem is gone, so i come here for help

**Comment by WindQAQ** [comment]
What's your TensorFlow version btw? Try to print `tf.__version__`.

**Comment by zhuimeng999** [comment]
sorry, i did not check the provided code carefully, i run the above code on the Ubuntu, there is no error! but i can confirm the problem exist because my origin code still product error and can not continue, but when i remove the line about image.transform, it can train normally.

my tensorflow version is 2.3.0

**Comment by WindQAQ** [comment]
It still seems not an addons' issue. Can you verify if the cublas install correctly? Maybe follow the link down below. Thanks!

https://stackoverflow.com/questions/52337791/verify-that-cublas-is-installed

**Comment by zhuimeng999** [comment]
the cublas is installed, but the cublas.h  location is different from the link, i found it in another system location
my arch machine use cuda 11.0, gtx 1660, the ubuntu use cuda 10.2, rtx2080

**Comment by zhuimeng999** [comment]
i checked the implement of image.transform, it is an tensorflow api wrapper.
my code can run ok with tensorflow 2.2 and addons v0.10.1,  i noticed that the old implement use addons own c++ kernel, and the tensorflow's implement seem's only used for some keras preprocessing module which may not require gradient

**Comment by WindQAQ** [comment]
Oh I see, have you tried to install addons from source? I guess you might use different cuda version for tensorflow and tensorflow-addons. A proper version should be `pip install tf-nightly` (this is compatible with cuda 11.0) and tfa from source with specifying
```
TF_CUDA_VERSION="11.0"
TF_CUDNN_VERSION="YOUR_CUDNN_VERSION"
```

**Comment by zhuimeng999** [comment]
the tensorflow and addons both installed from source

**Comment by WindQAQ** [comment]
> i checked the implement of image.transform, it is an tensorflow api wrapper.
> my code can run ok with tensorflow 2.2 and addons v0.10.1, i noticed that the old implement use addons own c++ kernel, and the tensorflow's implement seem's only used for some keras preprocessing module which may not require gradient

Nope, tensorflow also invokes underlying ImageProjectiveTransform c++ kernel too.

**Comment by WindQAQ** [comment]
Can you try `tensorflow==2.3.0` with `tfa-nightly`? (maybe in the virtualenv so it won't break your package management).

**Comment by zhuimeng999** [comment]
i just checked it out, but my driver version does not work with the default pip repo,  does the tfa-nightly is the latest master branch, i just build it from source, but the problem still exist

**Comment by zhuimeng999** [comment]
I just modified the TensorFlow code to get the explicit err code, what the cusolverDnCreate(&cusolver_dn_handle) return is 7

**Comment by bhack** [comment]
Is http://www.jcuda.org/jcuda/jcusolver/doc/jcuda/jcusolver/cusolverStatus.html#CUSOLVER_STATUS_INTERNAL_ERROR
>  To correct: check that the hardware, an appropriate version of the driver, and the cuSolver library are correctly installed.

**Comment by zhuimeng999** [comment]
the return code 7 is an impossible status for cusolverDnCreate (https://docs.nvidia.com/cuda/cusolver/index.html#cuSolverDNcreate), there may be some stack broken, and seems not addons bug, so i will not bother you any more
